src/lib.rs

```diff
@@ -38,17 +38,28 @@ pub enum LoadingError {
     Other(#[from] anyhow::Error),
 }
 
-#[derive(Deserialize)]
-struct RustdocFormatVersion {
-    format_version: u32,
-}
-
 fn detect_rustdoc_format_version(path: &Path, file_data: &str) -> Result<u32, LoadingError> {
-    let version = serde_json::from_str::<RustdocFormatVersion>(file_data).map_err(|e| {
-        LoadingError::RustdocFormatDetection(path.display().to_string(), anyhow::Error::from(e))
-    })?;
+    // The last characters of a rustdoc file are always: ,"format_version":NUM}
```

Comment:
We should probably have a test on this function that proves the version detection works correctly even if `format_version` is at the start of the file and the end of the file has some very large unrelated string-valued field.